lib/multipartform.js

```diff
@@ -108,17 +108,25 @@ Part.prototype = {
 
   	//Now write out the body of the Part
     if (this.value instanceof File) {
-  	  fs.open(this.value.path, "r", 0666, function (err, fd) {
+  	  fs.open(this.value.path, "r+", 0o666, function (err, fd) {
     	  if (err) throw err;
 
   		  var position = 0;
+        var stats = fs.fstatSync(fd);
```

Comment:
Can we use async? Otherwise, this stuff is complicated enough that we can leave it sync if we need to.